ref(alerts): Add docs for dynamic alerts (#11158)
getsentry/sentry-docs: docs/product/alerts/create-alerts/metric-alert-config.mdx

## Thresholds

There are three threshold types:

> - **Static**: A fixed threshold, such as when there are 100 errors in a period of time.

Reviewer: ```suggestion
- **Static**: A fixed numerical threshold set by you. (For example, if there are 100 errors in a set period of time.)
```

> - **Percent change**: A percent based threshold, such as when there are 10% more errors in a time period compared to a previous period. These are also referred to as [Change Alerts](#change-alerts-percent-change).

Reviewer 2: ```suggestion
- **Percent change**: A percent-based threshold, such as when there are 10% more errors in a time period compared to a previous period. These are also referred to as [Change Alerts](#change-alerts-percent-change).
```

> - **Percent change**: A percent-based threshold, such as when there are 10% more errors in a time period compared to a previous period. These are also referred to as [Change Alerts](#change-alerts-percent-change).

Reviewer: ```suggestion
- **Percent change**: A percent-based threshold set by you. (For example, if 10% more errors are detected in a time period compared to a previous period.) These are also referred to as [Change Alerts](#change-alerts-percent-change).
```

> - **Dynamic**: A dynamic threshold set by Sentry that detects anomalies whenever values fall outside expected bounds.

Reviewer: ```suggestion
- **Dynamic**: A dynamic threshold set by Sentry that detects anomalies whenever values fall outside of expected bounds.
```

![When the percent change option is selected.](./img/percent-change.png)

### Dynamic Alerts

<Include name="feature-available-for-user-group-early-adopter.mdx" />

> Dynamic alerts can be used when you don't know what threshold to set, but you know you want to be alerted when something is far outside the bounds of normalcy. Sentry will look at the historical data for the given metric and determine if the current data is anomalous. You can select how responsive the alert will be, and whether you want to be alerted when the metric is above, below, or both above and below the expected bounds.

Reviewer 2: ```suggestion
Dynamic alerts can be used when you don't know what threshold to set, but you know you want to be alerted when something is far outside the bounds of normalcy. Sentry will look at the historical data for the given metric and determine if the current data is anomalous. You can select how responsive the alert will be, and whether you want to be alerted when the metric is above and/or below the expected bounds.
```

> Dynamic alerts can be used when you don't know what threshold to set, but you know you want to be alerted when something is far outside the bounds of normalcy. Sentry will look at the historical data for the given metric and determine if the current data is anomalous. You can select how responsive the alert will be, and whether you want to be alerted when the metric is above and/or below the expected bounds.
> 
> Dynamic alerts do not use the Critical, Warning, and Resolved thresholds. When no more anomalies are detected, the alert will resolve on its own.

Reviewer: ```suggestion
Dynamic alerts can be used when you're not sure about what threshold to set, but want to be alerted when something is far outside the bounds of normalcy. To determine what is and isn't normal, Sentry will look at the historical data for the given metric and compare it to the current data to see if it's anomalous. You can select how responsive the alert will be, and whether you want to be alerted when the metric is above and/or below the expected bounds.

Dynamic alerts don't use the Critical, Warning, and Resolved thresholds. When no more anomalies are detected, the alert will resolve on its own.
```

> Dynamic alerts do not use the Critical, Warning, and Resolved thresholds like other alert threshold types as you are not setting these yourself. When no more anomalies are detected, the alert will resolve.

Reviewer 2: I'm not sure why it matters that "you're not setting these yourself". Can you rephrase what you're trying to get across here or maybe we can just remove it so it's easier to read?

```suggestion
Dynamic alerts do not use the Critical, Warning, and Resolved thresholds. When no more anomalies are detected, the alert will resolve on its own.
```
